test/e2e/utils/secrets.go

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// createSecret creates a secret in the specified namespace from the given cert and key files.
+func createSecret(ctx context.Context, cl client.Client, name, namespace, certsDir, certFileName, keyFileName, certDataKey, keyDataKey string) error {
+	cert, err := os.ReadFile(filepath.Join(certsDir, certFileName)) // #nosec: G304 -- test files.
+	if err != nil {
+		return err
+	}
+	key, err := os.ReadFile(filepath.Join(certsDir, keyFileName)) // #nosec: G304 -- test files.
+	if err != nil {
+		return err
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			certDataKey: cert,
+			keyDataKey:  key,
+		},
+	}
+	return cl.Create(ctx, secret)
+}
+
+func CreateCASecret(ctx context.Context, cl client.Client, name, namespace, certsDir string) error {
+	return createSecret(ctx, cl, name, namespace, certsDir, caCertFileName, caKeyFileName, "ca.crt", "ca.key")
+}
+
+func CreateServerTLSSecret(ctx context.Context, cl client.Client, name, namespace, certsDir string) error {
```

Comment:
Nit: missing function comment (for consistency)
```suggestion
// CreateServerTLSSecret creates the secret containing the server TLS certificates.
func CreateServerTLSSecret(ctx context.Context, cl client.Client, name, namespace, certsDir string) error {
```